File: hw/pazmino/u3/HW30URIs/HW30URI/src/main/java/ec/edu/espe/hw30uri/controller/ProjectController.java
package ec.edu.espe.hw30uri.controller;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
import org.bson.conversions.Bson;
import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Updates.set;
import utils.MongoDBConnection;
import ec.edu.espe.hw30uri.model.Project;

/**
 *
 * @author Brandon Pazmino
 */
public class ProjectController {

    private static final String COLLECTION_NAME = "Projects";
    private static MongoCollection<Document> collection = MongoDBConnection.getDatabase().getCollection(COLLECTION_NAME);

    public static Document getProject(String projectId) {
        return collection.find(eq("projectId", projectId)).first();
    }

    public static boolean addProject(Project project) {
        try {
            collection.insertOne(project.toDocument());
            return true;
        } catch (Exception e) {
            System.out.println("Error al agregar el proyecto: " + e.getMessage());
            return false;
        }
    }

    public static boolean updateProjectStatus(String projectId, String newStatus) {
        Bson filter = eq("projectId", projectId);
        Bson update = set("operationalStatus", newStatus);
        return collection.updateOne(filter, update).getModifiedCount() > 0;
    }

    public static boolean deleteProject(String projectId) {
        return collection.deleteOne(eq("projectId", projectId)).getDeletedCount() > 0;
    }

}

File: hw/pazmino/u3/HW30URIs/hw40URIPython/ec/espe/espe/hw40_uri_python/controller/project_controller.py
from pymongo import MongoClient
from pymongo.collection import Collection
from pymongo.database import Database
from bson import ObjectId
from ec.espe.espe.hw40_uri_python.model import project
from utils.mongodb_connection import MongoDBConnection

class ProjectController:
    COLLECTION_NAME = "Projects"
    collection: Collection = MongoDBConnection.get_database()[COLLECTION_NAME]

    @staticmethod
    def get_project(project_id: str):
        return ProjectController.collection.find_one({"projectId": project_id})

    @staticmethod
    def add_project(project: project) -> bool:
        try:
            ProjectController.collection.insert_one(project.to_document())
            return True
        except Exception as e:
            print(f"Error al agregar el proyecto: {e}")
            return False

    @staticmethod
    def update_project_status(project_id: str, new_status: str) -> bool:
        result = ProjectController.collection.update_one(
            {"projectId": project_id},
            {"$set": {"operationalStatus": new_status}}
        )
        return result.modified_count > 0

    @staticmethod
    def delete_project(project_id: str) -> bool:
        result = ProjectController.collection.delete_one({"projectId": project_id})
        return result.deleted_count > 0
